tests/schema/test_generics_interface.py

```diff
@@ -0,0 +1,60 @@
+import textwrap
+from typing import Generic, TypeVar
+
+import strawberry
+
+
+def test_supports_generic_interface():
+    T = TypeVar("T")
+
+    @strawberry.interface
+    class Example(Generic[T]):
+        cursor: strawberry.ID
+        field: T
+
+    @strawberry.type
+    class User(Example[str]):
+        ...
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def example(self) -> User:
```

Comment:
so the use case I had in mind was doing something like this:

```suggestion
        def example(self) -> Example:
```

so that the generated schema would be:

```graphql
        type Query {
          example: StrExample!
        }

        interface StrExample {
          cursor: ID!
          field: String!
        }

        type User implements StrExample {
          cursor: ID!
          field: String!
        }
```

I tested it out and it doesn't work for my schema – raises an exception like this:

```
TypeError: ... fields cannot be resolved. The type "..." is generic, but no type has 
been passed
```

I wonder if what I'm after is even valid GraphQL? 🤔 I hope so!

Reply:
Also, in my case the generic field would be private:

```
field: strawberry.Private[T]
```

 so there is no need to expose that type and create an instantiation of the generic

Reply:
thanks! having just private be generic makes things easier in a way, because we don't really need to make a GraphQL copy of the type, I actually just pushed a comment for similar case, but let me add a test for what you need :)

Reply:
Amazing, thank you!

Reply:
> thanks! having just private be generic makes things easier in a way, because we don't really need to make a GraphQL copy of the type

Yup! I also think that it's reasonable to just fail in the non-private case.

Looking at the example below IMO it's impossible to tell what the desired GraphQL type should be. We don't know the type of the `field` field.

```python
    @strawberry.interface
    class Example(Generic[T]):
        cursor: strawberry.ID
        field: T

    @strawberry.type
    class Query:
        @strawberry.field
        def example(self) -> Example:
            ...
```

Reply:
Similarly, I think it's okay for this to result in a failure as well:

```python
    @strawberry.interface
    class Example(Generic[T]):
        cursor: strawberry.ID
        field: T

    @strawberry.type
    class Query:
        @strawberry.field
        def example(self) -> Example[Any]:
            ...
```

because I don't think it's possible to generate a GraphQL type like this:

```graphql
        interface AnyExample {
          cursor: ID!
          field: Any! # there is no Any type in GraphQL
        }
```

Reply:
I think both cases would already fail (at least when not used with interfaces) 😊